Editor/Scripts/BotManagement/RGBotSynchronizer.cs

```diff
@@ -38,13 +38,47 @@ private static string ZipPathForBot(long botId)
         [MenuItem("Regression Games/Create New Bot")]
         private static void CreateNewBot()
         {
-            var botId = RGSettings.GetOrCreateSettings().GetNextBotId();
+            // Create our bot folder if it doesn't exist
+            RGEditorUtils.CreateAllAssetFolders(BOTS_PATH);
+
+            // prompt the user to choose a folder
+            var newBotPath = EditorUtility.SaveFilePanelInProject(
```

Comment:
If I already have `MyCoolBot` and I try to create a second `MyCoolBot` using the menu, where does it handle it ?  Does this file panel itself stop you, or do we have to handle it in code somewhere ?

Reply:
That experience is a little less ideal than I'd like, but it's functional. I was hoping we could check for that case and provide an error or alternate behavior. In my testing, what happens in Unity pops a dialog:

<img width="293" alt="image" src="https://github.com/Regression-Games/RGUnityBots/assets/7574/1c284259-476d-4fd9-bb7b-4190ef7104e1">

This seems fine, but unfortunately _both_ buttons do the same thing, which is to return to the Save dialog. That's not terrible because at least it doesn't actually overwrite any bot code, and it puts control back in the user's hands. It's kinda annoying that the "Replace" button doesn't actually dismiss the dialog though, we could handle that case in our code if Unity told us it was happening. I think it's because this is the _Save File_ dialog but the conflict is with a folder. We're kinda abusing this dialog a _tiny_ bit because we're prompting to save a file but actually creating a folder. The other dialogs don't work for this though, the only one related to folders allows you to pick a folder to save something in, but doesn't give you control over the name.

So, I think the current behavior is OK, if not ideal.